src/Nethermind/Nethermind.Merge.Plugin/Synchronization/BeaconHeadersSyncFeed.cs

```diff
@@ -111,32 +133,29 @@ protected override AddBlockResult InsertToBlockTree(BlockHeader header)
         {
             options |= BlockTreeInsertOptions.TotalDifficultyNotNeeded;
         }
-
-        AddBlockResult insertOutcome = _blockTree.IsKnownBlock(header.Number, header.Hash)
-            ? AddBlockResult.AlreadyKnown
-            : _blockTree.Insert(header, options);
+        
         // Found existing block in the block tree
-        if (insertOutcome == AddBlockResult.AlreadyKnown)
+        if (_blockTree.IsKnownBlock(header.Number, header.Hash))
         {
-            // if (_blockTree.LowestInsertedHeader != null
-            //     && _blockTree.LowestInsertedHeader.Number < (_blockTree.LowestInsertedBeaconHeader?.Number ?? long.MaxValue))
-            // {
-            if (_logger.IsTrace)
-                _logger.Trace(
-                    " BeaconHeader LowestInsertedBeaconHeader found existing chain in fast sync," +
-                    $"old: {_blockTree.LowestInsertedBeaconHeader?.Number}, new: {_blockTree.LowestInsertedHeader.Number}");
-            // beacon header set to (global) lowest inserted header
-            //   _blockTree.LowestInsertedBeaconHeader = _blockTree.LowestInsertedHeader;
+            if (_blockTree.LowestInsertedBeaconHeader?.ParentHash == header.Hash)
```

Comment:
I don't understand, if the header feed feed the header in reverse (head to root), then shouldn't the previous insterted block here always be the LowestInsertedBeaconHeader, and it's parent always be the current `header` and therefore, this block will always be true?